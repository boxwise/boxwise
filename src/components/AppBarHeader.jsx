import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { Link } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import ArrowBack from "@material-ui/icons/ArrowBack";
import Typography from "@material-ui/core/Typography";

const styles = theme => ({
  titleText: {
    [theme.breakpoints.up("sm")]: {
      margin: "0 auto"
    }
  }
});

export const AppBarHeader = ({ classes, title, backUrl }) => (
  <AppBar position="fixed">
    <Toolbar>
      <IconButton
        color="inherit"
        aria-label="go back"
        component={Link}
        to={backUrl}
      >
        <ArrowBack />
      </IconButton>
      <Typography
        className={classes.titleText}
        variant="h6"
        color="inherit"
        noWrap
      >
        {title}
      </Typography>
    </Toolbar>
  </AppBar>
);
export default withStyles(styles)(AppBarHeader);
