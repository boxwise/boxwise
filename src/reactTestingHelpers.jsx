import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent } from "react-testing-library";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import thunk from "redux-thunk";
import { createStore, applyMiddleware } from "redux";

export { fireEvent, wait } from "react-testing-library";

// following suggestions in https://testing-library.com/docs/react-testing-library/setup#custom-render

export const typeText = (element, value) => {
  fireEvent.change(element, { target: { value } });
};

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
});

export function customRender(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {},
    reducer = state => state
  } = {}
) {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    </MuiThemeProvider>
  );
}

export { customRender as render };
