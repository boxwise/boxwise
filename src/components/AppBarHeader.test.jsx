import React from "react";

import { render } from "reactTestingHelpers";

import { AppBarHeader } from "./AppBarHeader";

describe("<AppBarHeader />", () => {
  it("renders correctly", () => {
    const { container } = render(
      <AppBarHeader title="Hello" backUrl="/dummy" classes="{}" />
    );
    expect(container).toMatchSnapshot();
  });
});
