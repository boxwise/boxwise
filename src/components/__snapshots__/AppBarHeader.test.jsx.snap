// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBarHeader /> renders correctly 1`] = `
<div>
  <header
    class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-8 mui-fixed"
  >
    <div
      class="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
      <a
        aria-label="go back"
        class="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42"
        href="/dummy"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label-46"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-50"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-95"
        />
      </a>
      <h6
        class="MuiTypography-root-59 MuiTypography-h6-76 MuiTypography-colorInherit-88 MuiTypography-noWrap-85"
      >
        Hello
      </h6>
    </div>
  </header>
</div>
`;
